ImageMagick uses many unsupported attributes by Oracle Solaris Studio.
Supported attributes are here:
http://docs.oracle.com/cd/E24457_01/html/E21991/gljol.html
================================================================================
--- ImageMagick-6.7.8/magick/method-attribute.h.orig	Sun Feb 19 10:15:53 2012
+++ ImageMagick-6.7.8/magick/method-attribute.h	Tue Oct 16 06:58:12 2012
@@ -110,7 +110,7 @@
 
 #if defined(MAGICKCORE_HAVE___ATTRIBUTE__)
 #  define magick_aligned(x)  __attribute__((aligned(x)))
-#  define magick_attribute  __attribute__
+#  define magick_attribute(x)
 #  define magick_unused(x)  magick_unused_ ## x __attribute__((unused))
 #else
 #  define magick_aligned(x)  /* nothing */
